### PR TITLE
Use location code to determine if a request is from BorrowDirect

### PR DESCRIPTION
**Pull request by marlo-longley**
Based on #833

Wanted to get this up while I was on the topic. However, I'm not sure of any current requests with this code to test with in the UI and will keep this as a draft until I test that.

### CODE REVIEW

File: app/models/folio/request.rb [hunk 1]
```diff
@@ -208,8 +204,7 @@ def from_ilb?
     end
 
     def from_borrow_direct?
-      # TODO
-      nil
+      library_key == 'SUL-BORROW-DIRECT'
```
Comment (cbeer) [inline app/models/folio/request.rb:207]: I thought this would probably be a location code, not a whole library?

**Comment by marlo-longley** [comment]
@cbeer this should make more sense now, thank you.

File: app/models/folio/request.rb [hunk 2]
```diff
@@ -99,7 +99,7 @@ def placed_library
 
     def library
       if from_borrow_direct?
-        Settings.BORROW_DIRECT_CODE
```
Comment (cbeer) [inline app/models/folio/request.rb:102]: Shouldn't we keep `Settings.BORROW_DIRECT_CODE` to make sure we trigger the right behaviors in the view?

Comment (marlo-longley) [reply app/models/folio/request.rb:102]: I'm not sure. I thought we wanted to stop using the Settings.

Comment (cbeer) [reply app/models/folio/request.rb:102]: We're currently using `BORROW_DIRECT` to get the right name mapping here:

https://github.com/sul-dlss/mylibrary/blob/main/config/application.rb#L65

Are we ready to get those mappings from FOLIO now?

Comment (marlo-longley) [reply app/models/folio/request.rb:102]: Ah. Thank you I didn't know this.

Comment (marlo-longley) [reply app/models/folio/request.rb:102]: Would it be appropriate to add "SUL-BORROW-DIRECT" as a mapping and make a note to remove the old one after FOLIO?

Comment (cbeer) [reply app/models/folio/request.rb:102]: If it were me, I'd keep using the `Settings` value and post-FOLIO we can just use the FOLIO discovery labels.

File: config/application.rb [hunk 3]
```diff
@@ -90,6 +89,9 @@ class Application < Rails::Application
       'SCIENCE' => 'Science Library (Li and Ma)',
       'SPEC-COLL' => 'Special Collections',
       'SPEC-DESK' => 'Special Collections',
+      # TODO: remove this first entry BORROW_DIRECT after FOLIO. SUL-BORROW-DIRECT is the new FOLIO code.
```
Comment (jcoyne) [inline config/application.rb:92]:
```suggestion
      # TODO: remove this first entry BORROW_DIRECT after migrating off of Symphony. SUL-BORROW-DIRECT is the new FOLIO code.
```

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: config/application.rb (hunk 3 above)
Comment (marlo-longley) [reply config/application.rb:92]: I will add this. The ILL code too will be replaced but can leave off that comment until we are addressing that particular code.

**Comment by marlo-longley** [comment]
@jcoyne I added your suggestion for the comment.

**Comment by jcoyne** [comment]
@marlo-longley I see you changed `settings.yml`, but I asked for the change in `config/application.rb`.

**Review by jcoyne — APPROVED** [review]

**Comment by jcoyne** [comment]
@marlo-longley please fix rubocop error and then feel free to merge.

**Review by jcoyne — APPROVED** [review]

**Pull request merged**